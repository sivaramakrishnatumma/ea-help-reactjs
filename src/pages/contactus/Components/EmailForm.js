import React, { Component } from "react";
import { connect } from "react-redux";
import Button from "../../../shared/components/UIElements/Button/Button";
import { sendEmail } from "../../../actions/actions";

class EmailForm extends Component {
  state = { formValues: {} };

  inputHandler = e => {
    let formValues = this.state.formValues;
    formValues[e.target.id] = e.target.value;
    this.setState({ formValues });
  };

  submitHandler = e => {
    e.preventDefault();
    const { product, platform, topic, issue } = this.props;
    const body = {
      product: product && product.name,
      platform: platform && platform.name,
      topic: topic && topic.name,
      issue: issue && issue.name,
      name: this.state.formValues.name,
      email: this.state.formValues.email,
      comments: this.state.formValues.comments
    };
    this.props.dispatch(sendEmail(body));
  };

  render() {
    return (
      <form onSubmit={this.submitHandler}>
        <div className="form-group">
          <label htmlFor="name">Name:</label>
          <input
            type="text"
            className="form-control"
            placeholder="Please enter your name"
            id="name"
            onInput={this.inputHandler}
          />
        </div>
        <div className="form-group">
          <label htmlFor="email">Email address:</label>
          <input
            type="email"
            className="form-control"
            placeholder="Please enter your email address"
            id="email"
            onInput={this.inputHandler}
          />
        </div>
        <div className="form-group">
          <label htmlFor="email">Comments:</label>
          <textarea
            className="form-control"
            placeholder="Write your comments"
            id="comments"
            onInput={this.inputHandler}
            rows="8"
          ></textarea>
        </div>
        <Button type="submit" className="btn btn-dark">
          Send Email
        </Button>
      </form>
    );
  }
}

const mapStateToProps = state => {
  // if (state.isEmailSent) {
  //   alert("Email Sent Successfully");
  // }
  return {
    ...state,
    product: state.selectedProduct,
    platform: state.selectedPlatform,
    topic: state.selectedTopic,
    issue: state.selectedIssue
  };
};

export default connect(mapStateToProps)(EmailForm);
