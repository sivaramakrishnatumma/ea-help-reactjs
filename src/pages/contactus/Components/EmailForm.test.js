import { mount } from "enzyme";
import React from "react";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import EmailForm from "./EmailForm";

describe("Email Form test case", () => {
  const mockStore = configureMockStore();
  let store = mockStore({});

  it("should test email form", () => {
    const initialState = {
      product: { name: "test-product" },
      platform: { name: "test-platform" },
      topic: { name: "test-topic" },
      issue: { name: "test-issue" }
    };
    store = mockStore(initialState);
    const wrapper = mount(
      <Provider store={store}>
        <EmailForm></EmailForm>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should test email form input change", () => {
    const initialState = {
      product: { name: "test-product" },
      platform: { name: "test-platform" },
      topic: { name: "test-topic" },
      issue: { name: "test-issue" }
    };
    store = mockStore(initialState);
    const inputHandler = jest.fn();
    const wrapper = mount(
      <Provider store={store}>
        <EmailForm></EmailForm>
      </Provider>
    );
    const event = { target: { value: "NewName" } };
    wrapper
      .find("input")
      .at(0)
      .simulate("input", event);
    expect(wrapper).toMatchSnapshot();
  });

  it("should test email form submit", () => {
    const props = {
      dispatch: jest.fn()
    };
    const initialState = {
      product: { name: "test-product" },
      platform: { name: "test-platform" },
      topic: { name: "test-topic" },
      issue: { name: "test-issue" }
    };
    store = mockStore(initialState);
    const inputHandler = jest.fn();
    const wrapper = mount(
      <Provider store={store}>
        <EmailForm {...props}></EmailForm>
      </Provider>
    );
    const event = { preventDefault: jest.fn() };
    wrapper
      .find("form")
      .at(0)
      .simulate("submit", event);
    expect(wrapper).toMatchSnapshot();
  });
});
