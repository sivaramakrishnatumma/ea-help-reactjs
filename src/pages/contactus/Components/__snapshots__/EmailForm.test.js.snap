// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email Form test case should test email form 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(EmailForm)>
    <EmailForm
      dispatch={[Function]}
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="name"
          >
            Name:
          </label>
          <input
            className="form-control"
            id="name"
            onInput={[Function]}
            placeholder="Please enter your name"
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="email"
          >
            Email address:
          </label>
          <input
            className="form-control"
            id="email"
            onInput={[Function]}
            placeholder="Please enter your email address"
            type="email"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="email"
          >
            Comments:
          </label>
          <textarea
            className="form-control"
            id="comments"
            onInput={[Function]}
            placeholder="Write your comments"
            rows="8"
          />
        </div>
        <Button
          className="btn btn-dark"
          type="submit"
        >
          <button
            className="btn btn btn-dark"
            type="submit"
          >
            Send Email
          </button>
        </Button>
      </form>
    </EmailForm>
  </Connect(EmailForm)>
</Provider>
`;

exports[`Email Form test case should test email form input change 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(EmailForm)>
    <EmailForm
      dispatch={[Function]}
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="name"
          >
            Name:
          </label>
          <input
            className="form-control"
            id="name"
            onInput={[Function]}
            placeholder="Please enter your name"
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="email"
          >
            Email address:
          </label>
          <input
            className="form-control"
            id="email"
            onInput={[Function]}
            placeholder="Please enter your email address"
            type="email"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="email"
          >
            Comments:
          </label>
          <textarea
            className="form-control"
            id="comments"
            onInput={[Function]}
            placeholder="Write your comments"
            rows="8"
          />
        </div>
        <Button
          className="btn btn-dark"
          type="submit"
        >
          <button
            className="btn btn btn-dark"
            type="submit"
          >
            Send Email
          </button>
        </Button>
      </form>
    </EmailForm>
  </Connect(EmailForm)>
</Provider>
`;

exports[`Email Form test case should test email form submit 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(EmailForm)
    dispatch={[MockFunction]}
  >
    <EmailForm
      dispatch={[Function]}
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="name"
          >
            Name:
          </label>
          <input
            className="form-control"
            id="name"
            onInput={[Function]}
            placeholder="Please enter your name"
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="email"
          >
            Email address:
          </label>
          <input
            className="form-control"
            id="email"
            onInput={[Function]}
            placeholder="Please enter your email address"
            type="email"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="email"
          >
            Comments:
          </label>
          <textarea
            className="form-control"
            id="comments"
            onInput={[Function]}
            placeholder="Write your comments"
            rows="8"
          />
        </div>
        <Button
          className="btn btn-dark"
          type="submit"
        >
          <button
            className="btn btn btn-dark"
            type="submit"
          >
            Send Email
          </button>
        </Button>
      </form>
    </EmailForm>
  </Connect(EmailForm)>
</Provider>
`;
